client/dashboard/profile-wizard/steps/theme/uploader.js

```diff
@@ -0,0 +1,117 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import classnames from 'classnames';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { DropZoneProvider, DropZone, Spinner } from '@wordpress/components';
+import Gridicon from 'gridicons';
+import { noop } from 'lodash';
+import PropTypes from 'prop-types';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Card, H } from '@woocommerce/components';
+
+class ThemeUploader extends Component {
+	constructor() {
+		super();
+
+		this.state = {
+			isUploading: false,
+		};
+
+		this.handleFilesDrop = this.handleFilesDrop.bind( this );
+	}
+
+	handleFilesDrop( files ) {
+		const file = files[ 0 ];
+
+		this.setState( { isUploading: true } );
+		this.uploadTheme( file );
+	}
+
+	uploadTheme( file ) {
+		const { addNotice, onUploadComplete } = this.props;
+
+		const body = new FormData();
+		body.append( 'pluginzip', file );
+
+		return apiFetch( { path: '/wc-admin/v1/themes', method: 'POST', body } )
+			.then( response => {
+				onUploadComplete( response );
+				this.setState( { isUploading: false } );
+				addNotice( { status: response.status, message: response.message } );
+			} )
+			.catch( error => {
+				this.setState( { isUploading: false } );
+				if ( error && error.message ) {
+					addNotice( { status: 'error', message: error.message } );
+				}
+			} );
+	}
+
+	render() {
+		const { className } = this.props;
+		const { isUploading } = this.state;
+
+		const classes = classnames( 'woocommerce-theme-uploader', className, {
+			'is-uploading': isUploading,
+		} );
+
+		return (
+			<Card className={ classes }>
+				<DropZoneProvider>
```

Comment:
I think we will want to combine this with a click handler as well. Users should be able to click to upload in addition to drag and drop.

Reply:
I agree.  Added a standard form file uploader to keep it accessible.